DDG4/edm4hep/Geant4Output2EDM4hep.cpp

```diff
@@ -467,14 +485,24 @@ void Geant4Output2EDM4hep::saveEvent(OutputContext<G4Event>& ctxt)  {
 
   // this does not compile as create() is we only get a const ref - need to review PODIO EventStore API
   edm4hep::EventHeaderCollection header_collection;
+
   auto header = header_collection.create();
   header.setRunNumber(runNumber);
   header.setEventNumber(eventNumber);
   header.setWeight(eventWeight);
   //not implemented in EDM4hep ?  header.setDetectorName(context()->detectorDescription().header().name());
-  header.setTimeStamp( std::time(nullptr) ) ;
-  m_frame.put( std::move(header_collection), "EventHeader");
+  header.setTimeStamp(std::time(nullptr));
+
+  // extract event header, in case we come from edm4hep input
+  auto* meh = context()->event().extension<edm4hep::MutableEventHeader>(false);
+  if(meh) {
+    header.setTimeStamp(meh->getTimeStamp());
+    for (auto const& weight: meh->getWeights()) {
+      header.addToWeights(weight);
+    }
```

Comment:
```suggestion
#if EDM4HEP_BUILD_VERSION >= EDM4HEP_VERSION(0, 99, 0)
    for (auto const& weight: meh->getWeights()) {
      header.addToWeights(weight);
    }
#endif
```

To not undo the efforts from #1380 again immediately ;)